### ISSUE TITLE
unable to find default formatters

### ISSUE THREAD

**Issue by Timothy-W-Hilton** [ryantam626/jupyterlab_code_formatter#255]
**Checklist prior to opening an issue**
- [x] I have followed fully the installation steps laid out in the documentation site.
- [x] I have restarted jupyterlab.
- [x] I have read the FAQ section in the documentation site.

**Describe the bug**
When I attempt to format a notebook or a cell, I get a dialog that says "JupyterLab Code Formatter Error \\ unable to find default formatters to use, please file an issue on GitHub".  I am able to run black successfully from my terminal and within my text editor.

**Diagnostic commands**
Please attach the output of the following commands (please format them properly)

- `pip freeze`

> anyio @ file:///home/conda/feedstock_root/build_artifacts/anyio_1641898043316/work/dist
> argon2-cffi @ file:///home/conda/feedstock_root/build_artifacts/argon2-cffi_1640817743617/work
> argon2-cffi-bindings @ file:///home/conda/feedstock_root/build_artifacts/argon2-cffi-bindings_1649500321618/work
> asttokens @ file:///home/conda/feedstock_root/build_artifacts/asttokens_1618968359944/work
> attrs @ file:///home/conda/feedstock_root/build_artifacts/attrs_1640799537051/work
> Babel @ file:///home/conda/feedstock_root/build_artifacts/babel_1619719576210/work
> backcall @ file:///home/conda/feedstock_root/build_artifacts/backcall_1592338393461/work
> backports.functools-lru-cache @ file:///home/conda/feedstock_root/build_artifacts/backports.functools_lru_cache_1618230623929/work
> beautifulsoup4 @ file:///home/conda/feedstock_root/build_artifacts/beautifulsoup4_1649463573192/work
> black @ file:///home/conda/feedstock_root/build_artifacts/black-recipe_1648499330704/work
> bleach @ file:///home/conda/feedstock_root/build_artifacts/bleach_1649361991009/work
> bokeh @ file:///home/conda/feedstock_root/build_artifacts/bokeh_1649828364523/work
> brotlipy @ file:///home/conda/feedstock_root/build_artifacts/brotlipy_1648854164373/work
> Cartopy @ file:///home/conda/feedstock_root/build_artifacts/cartopy_1646828066549/work
> certifi==2021.10.8
> cffi @ file:///home/conda/feedstock_root/build_artifacts/cffi_1625835307225/work
> cftime @ file:///home/conda/feedstock_root/build_artifacts/cftime_1649636863755/work
> chardet @ file:///home/conda/feedstock_root/build_artifacts/chardet_1649184154184/work
> charset-normalizer @ file:///home/conda/feedstock_root/build_artifacts/charset-normalizer_1644853463426/work
> click @ file:///home/conda/feedstock_root/build_artifacts/click_1648793473567/work
> cloudpickle @ file:///home/conda/feedstock_root/build_artifacts/cloudpickle_1631273254894/work
> colorama @ file:///home/conda/feedstock_root/build_artifacts/colorama_1602866480661/work
> colorcet @ file:///home/conda/feedstock_root/build_artifacts/colorcet_1638280441091/work
> conda==4.12.0
> conda-build==3.21.8
> conda-package-handling @ file:///home/conda/feedstock_root/build_artifacts/conda-package-handling_1649385069444/work
> cryptography @ file:///home/conda/feedstock_root/build_artifacts/cryptography_1649035236822/work
> cycler @ file:///home/conda/feedstock_root/build_artifacts/cycler_1635519461629/work
> cytoolz==0.11.2
> dask @ file:///home/conda/feedstock_root/build_artifacts/dask-core_1650069300515/work
> dataclasses @ file:///home/conda/feedstock_root/build_artifacts/dataclasses_1628958434797/work
> datashader @ file:///home/conda/feedstock_root/build_artifacts/datashader_1651003146494/work
> datashape==0.5.4
> debugpy @ file:///home/conda/feedstock_root/build_artifacts/debugpy_1649586347330/work
> decorator @ file:///home/conda/feedstock_root/build_artifacts/decorator_1641555617451/work
> defusedxml @ file:///home/conda/feedstock_root/build_artifacts/defusedxml_1615232257335/work
> distributed @ file:///home/conda/feedstock_root/build_artifacts/distributed_1650076440351/work
> entrypoints @ file:///home/conda/feedstock_root/build_artifacts/entrypoints_1643888246732/work
> executing @ file:///home/conda/feedstock_root/build_artifacts/executing_1646044401614/work
> fastjsonschema @ file:///home/conda/feedstock_root/build_artifacts/python-fastjsonschema_1641751198313/work/dist
> filelock @ file:///home/conda/feedstock_root/build_artifacts/filelock_1645136850180/work
> flit_core @ file:///home/conda/feedstock_root/build_artifacts/flit-core_1645629044586/work/source/flit_core
> fonttools @ file:///home/conda/feedstock_root/build_artifacts/fonttools_1651017733844/work
> fsspec @ file:///home/conda/feedstock_root/build_artifacts/fsspec_1648820545323/work
> geopandas @ file:///home/conda/feedstock_root/build_artifacts/geopandas_1638382567988/work
> geoviews @ file:///home/conda/feedstock_root/build_artifacts/geoviews-core_1646773856131/work
> glob2==0.7
> HeapDict==1.0.1
> holoviews @ file:///home/conda/feedstock_root/build_artifacts/holoviews_1645018991041/work
> hvplot @ file:///home/conda/feedstock_root/build_artifacts/hvplot_1627048757529/work
> idna @ file:///home/conda/feedstock_root/build_artifacts/idna_1642433548627/work
> importlib-metadata @ file:///home/conda/feedstock_root/build_artifacts/importlib-metadata_1648728288044/work
> importlib-resources @ file:///home/conda/feedstock_root/build_artifacts/importlib_resources_1650169872758/work
> iniconfig @ file:///home/conda/feedstock_root/build_artifacts/iniconfig_1603384189793/work
> ipykernel @ file:///home/conda/feedstock_root/build_artifacts/ipykernel_1649684276995/work/dist/ipykernel-6.13.0-py3-none-any.whl
> ipython @ file:///home/conda/feedstock_root/build_artifacts/ipython_1648413562175/work
> ipython-genutils==0.2.0
> isort @ file:///home/conda/feedstock_root/build_artifacts/isort_1636447814597/work
> jedi @ file:///home/conda/feedstock_root/build_artifacts/jedi_1649067096717/work
> Jinja2 @ file:///home/conda/feedstock_root/build_artifacts/jinja2_1648299710939/work
> json5 @ file:///home/conda/feedstock_root/build_artifacts/json5_1600692310011/work
> jsonschema @ file:///home/conda/feedstock_root/build_artifacts/jsonschema-meta_1642000296051/work
> jupyter-client @ file:///home/conda/feedstock_root/build_artifacts/jupyter_client_1650972844302/work
> jupyter-core @ file:///home/conda/feedstock_root/build_artifacts/jupyter_core_1645024267147/work
> jupyter-server @ file:///home/conda/feedstock_root/build_artifacts/jupyter_server_1651092495905/work
> jupyterlab @ file:///home/conda/feedstock_root/build_artifacts/jupyterlab_1650038326843/work
> jupyterlab-code-formatter @ file:///home/conda/feedstock_root/build_artifacts/jupyterlab_code_formatter_1617908844240/work
> jupyterlab-pygments @ file:///home/conda/feedstock_root/build_artifacts/jupyterlab_pygments_1649936611996/work
> jupyterlab-server @ file:///home/conda/feedstock_root/build_artifacts/jupyterlab_server_1650422149061/work
> kiwisolver @ file:///home/conda/feedstock_root/build_artifacts/kiwisolver_1648854392795/work
> libarchive-c @ file:///home/conda/feedstock_root/build_artifacts/python-libarchive-c_1649436017970/work
> llvmlite==0.36.0
> locket @ file:///home/conda/feedstock_root/build_artifacts/locket_1650660393415/work
> lz4 @ file:///home/conda/feedstock_root/build_artifacts/lz4_1649094500002/work
> Markdown @ file:///home/conda/feedstock_root/build_artifacts/markdown_1637220118004/work
> MarkupSafe @ file:///home/conda/feedstock_root/build_artifacts/markupsafe_1648737556467/work
> matplotlib @ file:///home/conda/feedstock_root/build_artifacts/matplotlib-suite_1639359646028/work
> matplotlib-inline @ file:///home/conda/feedstock_root/build_artifacts/matplotlib-inline_1631080358261/work
> mistune @ file:///home/conda/feedstock_root/build_artifacts/mistune_1635844679781/work
> msgpack @ file:///home/conda/feedstock_root/build_artifacts/msgpack-python_1648745988851/work
> multipledispatch==0.6.0
> munkres==1.1.4
> mypy-extensions @ file:///home/conda/feedstock_root/build_artifacts/mypy_extensions_1649013330831/work
> nbclassic @ file:///home/conda/feedstock_root/build_artifacts/nbclassic_1647450696711/work
> nbclient @ file:///home/conda/feedstock_root/build_artifacts/nbclient_1649785910383/work
> nbconvert @ file:///home/conda/feedstock_root/build_artifacts/nbconvert-meta_1649676641343/work
> nbformat @ file:///home/conda/feedstock_root/build_artifacts/nbformat_1649105480291/work
> nest-asyncio @ file:///home/conda/feedstock_root/build_artifacts/nest-asyncio_1648959695634/work
> netCDF4 @ file:///home/conda/feedstock_root/build_artifacts/netcdf4_1636302994507/work
> notebook @ file:///home/conda/feedstock_root/build_artifacts/notebook_1650363291341/work
> notebook-shim @ file:///home/conda/feedstock_root/build_artifacts/notebook-shim_1646330736330/work
> numba @ file:///home/conda/feedstock_root/build_artifacts/numba_1623568543876/work
> numpy @ file:///home/conda/feedstock_root/build_artifacts/numpy_1651020388495/work
> packaging @ file:///home/conda/feedstock_root/build_artifacts/packaging_1637239678211/work
> pandas==1.4.2
> pandocfilters @ file:///home/conda/feedstock_root/build_artifacts/pandocfilters_1631603243851/work
> panel @ file:///home/conda/feedstock_root/build_artifacts/panel_1650330163452/work
> param @ file:///home/conda/feedstock_root/build_artifacts/param_1648753599418/work
> parso @ file:///home/conda/feedstock_root/build_artifacts/parso_1638334955874/work
> partd @ file:///home/conda/feedstock_root/build_artifacts/partd_1617910651905/work
> pathspec @ file:///home/conda/feedstock_root/build_artifacts/pathspec_1626613672358/work
> patsy @ file:///home/conda/feedstock_root/build_artifacts/patsy_1632667180946/work
> pexpect @ file:///home/conda/feedstock_root/build_artifacts/pexpect_1602535608087/work
> pickleshare @ file:///home/conda/feedstock_root/build_artifacts/pickleshare_1602536217715/work
> Pillow @ file:///home/conda/feedstock_root/build_artifacts/pillow_1649817996082/work
> pkginfo @ file:///home/conda/feedstock_root/build_artifacts/pkginfo_1638813452194/work
> platformdirs @ file:///home/conda/feedstock_root/build_artifacts/platformdirs_1645298319244/work
> pluggy @ file:///home/conda/feedstock_root/build_artifacts/pluggy_1648772597647/work
> prometheus-client @ file:///home/conda/feedstock_root/build_artifacts/prometheus_client_1649447152425/work
> prompt-toolkit @ file:///home/conda/feedstock_root/build_artifacts/prompt-toolkit_1649130487073/work
> psutil @ file:///home/conda/feedstock_root/build_artifacts/psutil_1648872973590/work
> ptyprocess @ file:///home/conda/feedstock_root/build_artifacts/ptyprocess_1609419310487/work/dist/ptyprocess-0.7.0-py2.py3-none-any.whl
> pure-eval @ file:///home/conda/feedstock_root/build_artifacts/pure_eval_1642875951954/work
> py @ file:///home/conda/feedstock_root/build_artifacts/py_1636301881863/work
> pycosat @ file:///home/conda/feedstock_root/build_artifacts/pycosat_1649384802055/work
> pycparser @ file:///home/conda/feedstock_root/build_artifacts/pycparser_1636257122734/work
> pyct==0.4.6
> Pygments @ file:///home/conda/feedstock_root/build_artifacts/pygments_1650904496387/work
> pyOpenSSL @ file:///home/conda/feedstock_root/build_artifacts/pyopenssl_1643496850550/work
> pyparsing @ file:///home/conda/feedstock_root/build_artifacts/pyparsing_1649603503565/work
> pyproj @ file:///home/conda/feedstock_root/build_artifacts/pyproj_1650803098020/work
> pyrsistent @ file:///home/conda/feedstock_root/build_artifacts/pyrsistent_1649013355209/work
> pyshp @ file:///home/conda/feedstock_root/build_artifacts/pyshp_1644593068012/work
> PySocks @ file:///home/conda/feedstock_root/build_artifacts/pysocks_1648857263093/work
> pytest==7.1.2
> python-dateutil @ file:///home/conda/feedstock_root/build_artifacts/python-dateutil_1626286286081/work
> pytz @ file:///home/conda/feedstock_root/build_artifacts/pytz_1647961439546/work
> pyviz-comms @ file:///home/conda/feedstock_root/build_artifacts/pyviz_comms_1648629824948/work
> PyYAML @ file:///home/conda/feedstock_root/build_artifacts/pyyaml_1648757097602/work
> pyzmq @ file:///home/conda/feedstock_root/build_artifacts/pyzmq_1649055645150/work
> requests @ file:///home/conda/feedstock_root/build_artifacts/requests_1641580202195/work
> ruamel-yaml-conda @ file:///home/conda/feedstock_root/build_artifacts/ruamel_yaml_1636009144459/work
> scipy @ file:///home/conda/feedstock_root/build_artifacts/scipy_1644357253444/work
> seaborn @ file:///home/conda/feedstock_root/build_artifacts/seaborn-split_1629095986539/work
> Send2Trash @ file:///home/conda/feedstock_root/build_artifacts/send2trash_1628511208346/work
> Shapely @ file:///home/conda/feedstock_root/build_artifacts/shapely_1649587720336/work
> six @ file:///home/conda/feedstock_root/build_artifacts/six_1620240208055/work
> sniffio @ file:///home/conda/feedstock_root/build_artifacts/sniffio_1648819188507/work
> sortedcontainers @ file:///home/conda/feedstock_root/build_artifacts/sortedcontainers_1621217038088/work
> soupsieve @ file:///home/conda/feedstock_root/build_artifacts/soupsieve_1638550740809/work
> stack-data @ file:///home/conda/feedstock_root/build_artifacts/stack_data_1644872665635/work
> statsmodels @ file:///home/conda/feedstock_root/build_artifacts/statsmodels_1644535581977/work
> tblib @ file:///home/conda/feedstock_root/build_artifacts/tblib_1616261298899/work
> terminado @ file:///home/conda/feedstock_root/build_artifacts/terminado_1649103779998/work
> tinycss2 @ file:///home/conda/feedstock_root/build_artifacts/tinycss2_1637612658783/work
> tomli @ file:///home/conda/feedstock_root/build_artifacts/tomli_1644342247877/work
> toolz @ file:///home/conda/feedstock_root/build_artifacts/toolz_1636177962100/work
> tornado @ file:///home/conda/feedstock_root/build_artifacts/tornado_1648827245914/work
> tqdm @ file:///home/conda/feedstock_root/build_artifacts/tqdm_1649051611147/work
> traitlets @ file:///home/conda/feedstock_root/build_artifacts/traitlets_1635260543454/work
> typed-ast @ file:///home/conda/feedstock_root/build_artifacts/typed-ast_1650218769046/work
> typing_extensions @ file:///home/conda/feedstock_root/build_artifacts/typing_extensions_1650370875435/work
> unicodedata2 @ file:///home/conda/feedstock_root/build_artifacts/unicodedata2_1649111919389/work
> urllib3 @ file:///home/conda/feedstock_root/build_artifacts/urllib3_1647489083693/work
> wcwidth @ file:///home/conda/feedstock_root/build_artifacts/wcwidth_1600965781394/work
> webencodings==0.5.1
> websocket-client @ file:///home/conda/feedstock_root/build_artifacts/websocket-client_1648562593984/work
> xarray @ file:///home/conda/feedstock_root/build_artifacts/xarray_1646254587173/work
> zict @ file:///home/conda/feedstock_root/build_artifacts/zict_1651156074437/work
> zipp @ file:///home/conda/feedstock_root/build_artifacts/zipp_1649012893348/work
> 

- `jupyter labextension list`

```
JupyterLab v3.3.4
/home/timh/mambaforge/envs/venv-urbanVPRM/share/jupyter/labextensions
        jupyterlab_pygments v0.2.2 enabled OK (python, jupyterlab_pygments)
        @pyviz/jupyterlab_pyviz v2.2.0 enabled OK (python, pyviz_comms)
        @ryantam626/jupyterlab_code_formatter v1.4.10 enabled OK (python, jupyterlab-code-formatter)

```

- `jupyter serverextension list`

```
config dir: /home/timh/.jupyter
    jupyterlab_code_formatter  enabled 
    - Validating...
      jupyterlab_code_formatter 1.4.10 OK
config dir: /home/timh/mambaforge/envs/venv-urbanVPRM/etc/jupyter
    jupyterlab  enabled 
    - Validating...
      jupyterlab 3.3.4 OK
    jupyterlab_code_formatter  enabled 
    - Validating...
      jupyterlab_code_formatter 1.4.10 OK
```

**Screenshots**

**Additional context**

Ubuntu 20.04.4 LTS

**Comment by ryantam626** [comment]
Hi!

Did you configure the plugin in any way? 

https://ryantam626.github.io/jupyterlab_code_formatter/_images/settings.gif
^ short gif on how you can check

**Comment by ryantam626** [comment]
Also what type of notebook is this?

**Comment by Timothy-W-Hilton** [comment]
Hi!  Thanks for your response.  So, I had looked over the configuration as mentioned in your comment above.  I wasn't really sure from [the docs](https://ryantam626.github.io/jupyterlab_code_formatter/configuration.html) what is supposed to go in there, so I hadn't mucked with it.  I just now tried typing "black" in the "python" box under "default_formatters" and "option 1", and now it's working!

I suppose it perhaps should have been obvious, but it wasn't to me.

ooooohkay, NOW I see the [Changing Default Formatters](https://ryantam626.github.io/jupyterlab_code_formatter /configuration.html#changing-default-formatter-s) section in the configuration docs.  I somehow managed to miss that on Friday.  

Thanks for this great tool!  It makes coding in a browser window much more palatable.
